File: ros2_python_tutorial/ros2_python_tutorial/p1_subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class MinimalSubscriber(Node):
    def __init__(self):
        super().__init__('minimal_subscriber')
        self.subscription_fast = self.create_subscription(String, 'fast_topic', self.listener_callback_fast, 10)
        self.subscription_fast
        self.subscription_slow = self.create_subscription(String, 'slow_topic', self.listener_callback_slow, 10)
        self.subscription_slow
    def listener_callback_fast(self, msg):
        self.get_logger().info('[Fast] I heard : "%s"' % msg.data)
    def listener_callback_slow(self, msg):
        self.get_logger().info('[Slow] I heard : "%s"' % msg.data)

def main(args=None):
    rclpy.init(args=args)
    minimal_subscriber = MinimalSubscriber()
    rclpy.spin(minimal_subscriber)
    minimal_subscriber.destroy_node()
    rclpy.shutdown()
if __name__ == '__main__':
    main()

File: ros2_cpp_tutorial/src/p1_subscriber.cpp
#include <memory>
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

using std::placeholders::_1;

class MinimalSubscriber : public rclcpp::Node {
    public:
        MinimalSubscriber() : Node("minimal_subscriber") {
            subscription_fast = this->create_subscription<std_msgs::msg::String>("fast_topic",10,std::bind(&MinimalSubscriber::topic_callback_fast, this, _1));
            subscription_slow = this->create_subscription<std_msgs::msg::String>("slow_topic",10,std::bind(&MinimalSubscriber::topic_callback_slow, this, _1));
        }
    private:
        void topic_callback_fast(const std_msgs::msg::String & msg) const {
            RCLCPP_INFO(this->get_logger(), "[Fast] I heard : '%s'",msg.data.c_str());
        }
        void topic_callback_slow(const std_msgs::msg::String & msg) const {
            RCLCPP_INFO(this->get_logger(), "[Slow] I heard : '%s'",msg.data.c_str());
        }
        rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_fast;
        rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_slow;
};
int main(int argc, char* argv[]) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<MinimalSubscriber>());
    rclcpp::shutdown();
    return 0;
}